### PR TITLE
domain: avoid panic in the init stats breaking the loadStatsWorker

### LINKED ISSUE #51581: panic in the init stats will break the loadStatsWorker
## Bug Report

Please answer these questions before submitting your issue. Thanks!

panic in the init stats will break the loadStatsWorker

### 1. Minimal reproduce step (Required)

loadStatsWorker has a recovery but the other subfunction in the loadStatsWorker have no recovery.

It is dangerouse.

image

image

### 2. What did you expect to see? (Required)

panic. loadStatsWorker can work
### 3. What did you see instead (Required)

panic. loadStatsWorker cannot work

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #51581

Problem Summary:

### What changed and how does it work?

If init-stats panics, then the load stats of this instance will all be broken.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/domain/domain.go [hunk 1]
```diff
@@ -2264,6 +2263,10 @@ func (do *Domain) newOwnerManager(prompt, ownerKey string) owner.Manager {
 func (do *Domain) initStats() {
 	statsHandle := do.StatsHandle()
 	defer func() {
+		if r := recover(); r != nil {
+			logutil.BgLogger().Error("panic when to init stats", zap.Any("r", r),
```
Comment (AilinKid) [inline pkg/domain/domain.go:2267]:
```suggestion
			logutil.BgLogger().Error("panic when initiating stats", zap.Any("r", r),
```

File: pkg/domain/domain.go [hunk 2]
```diff
@@ -303,8 +303,7 @@ func (do *Domain) loadInfoSchema(startTS uint64) (infoschema.InfoSchema, bool, i
 	if err != nil {
 		return nil, false, currentSchemaVersion, nil, err
 	}
-	// clear data
-	do.infoCache.Data = infoschema.NewData()
+
```
Comment (AilinKid) [inline pkg/domain/domain.go:306]: why remove this？

Comment (hawkingrei) [reply pkg/domain/domain.go]: It is some wrong commits. I revert it.

File: pkg/domain/domain.go (hunk 1 above)
Comment (hawkingrei) [reply pkg/domain/domain.go]: Done

**Review by AilinKid — APPROVED** [review]

**Review by elsa0520 — APPROVED** [review]
LGTM

**Review by zimulala — APPROVED** [review]

**Pull request merged**